delta-lake/src/integration/java/org/apache/iceberg/delta/TestSnapshotDeltaLakeTable.java

```diff
@@ -532,13 +533,14 @@ private void addExternalDatafiles(
   private static String getFullFilePath(String path, String tableRoot) {
     URI dataFileUri = URI.create(path);
     try {
-      String decodedPath = new URLCodec().decode(path);
+      String decodedPath = URLDecoder.decode(path, StandardCharsets.UTF_8.name());
       if (dataFileUri.isAbsolute()) {
         return decodedPath;
       } else {
         return tableRoot + File.separator + decodedPath;
       }
-    } catch (DecoderException e) {
+    } catch (UnsupportedEncodingException e) {
+      // In Java 10+ this will not need try-catch
```

Comment:
is that comment relevant? I think we can probably remove it

Reply:
Yes, I suppose that the tooling will catch this when the project gets upgraded to a newer Java version. 

Removed